### PR TITLE
Allow to increment strings

### LINKED ISSUE #174: FP: string++ | "Only numeric types are allowed in post-increment, string given."
# Bug report

I am uncertain if this is a "normal" use-case, but I see sometimes code like this: ```$a = 'a'; $a++;```

https://3v4l.org/JEv7V

... especially in some excel related code, this is working as expected. 

### Code snippet that reproduces the problem

https://phpstan.org/r/379c2ebf-cde5-4d10-aab2-17c9080a984a

### Expected output

no error for `ConstantStringType` [a-zA-Z]{0,1}

### Did PHPStan help you today? Did it make you happy in any way?

Currently, I am trying to enable the strict rules for a legacy project, and it's cool that we can enable one rules after another. Thanks for these features. :1st_place_medal:

### PR DESCRIPTION
**Pull request by VincentLanglet**
Closes https://github.com/phpstan/phpstan-strict-rules/issues/174

1.5.x seemed to be the default branch but I can also open the PR on 1.4.x

### CODE REVIEW

File: src/Rules/Operators/OperatorRuleHelper.php [hunk 1]
```diff
@@ -49,6 +49,11 @@ public function isValidForIncrementOrDecrement(Scope $scope, Expr $expr): bool
 			return true;
 		}
 
+		if ($type instanceof StringType) {
```
Comment (staabm) [inline src/Rules/Operators/OperatorRuleHelper.php]:
```suggestion
		if ($type->isString()->yes()) {
```

File: tests/Rules/Operators/OperandInArithmeticPostDecrementRuleTest.php [hunk 2]
```diff
@@ -22,10 +22,6 @@ protected function getExpectedErrors(): array
 				'Only numeric types are allowed in post-decrement, false given.',
 				21,
 			],
-			[
-				'Only numeric types are allowed in post-decrement, string given.',
```
Comment (ondrejmirtes) [inline tests/Rules/Operators/OperandInArithmeticPostDecrementRuleTest.php]: We need to keep this error. It doesn't do anything: https://3v4l.org/3b3hf

File: tests/Rules/Operators/OperandInArithmeticPreDecrementRuleTest.php [hunk 3]
```diff
@@ -22,10 +22,6 @@ protected function getExpectedErrors(): array
 				'Only numeric types are allowed in pre-decrement, false given.',
 				43,
 			],
-			[
-				'Only numeric types are allowed in pre-decrement, string given.',
```
Comment (ondrejmirtes) [inline tests/Rules/Operators/OperandInArithmeticPreDecrementRuleTest.php]: dtto https://3v4l.org/2KRC2

**Review by ondrejmirtes — CHANGES_REQUESTED** [review]

File: src/Rules/Operators/OperatorRuleHelper.php (hunk 1 above)
Comment (VincentLanglet) [reply src/Rules/Operators/OperatorRuleHelper.php]: Thanks, changed

File: tests/Rules/Operators/OperandInArithmeticPostDecrementRuleTest.php (hunk 2 above)
Comment (VincentLanglet) [reply tests/Rules/Operators/OperandInArithmeticPostDecrementRuleTest.php]: Indeed, I changed.

File: tests/Rules/Operators/OperandInArithmeticPreDecrementRuleTest.php (hunk 3 above)
Comment (VincentLanglet) [reply tests/Rules/Operators/OperandInArithmeticPreDecrementRuleTest.php]: Done

**Comment by ondrejmirtes** [comment]
Thank you.

**Pull request merged**